Document and fix

PaintNinja opened the pull request:
The current state of the repo lacks documentation and some required files to setup, test and contribute changes, as well as having some bugs that make testing harder (such as the use of `://` for some external scripts, which doesn't work when viewing files locally without having to muck about setting up a whole local web server).

This PR aims to fix that.

Note: Although it should be fine, it might be wise to do a server backup if you haven't already, just in case merging this breaks something I'm unaware of inside your server setup.

PaintNinja committed:
Start documenting and fixing

- Added Instructions.md with some more detailed instructions on how to work with this repo
- Updated the README.md to mention the new Instructions.md and link to it
- Provide a global_overrides.json that pagegen needs in order to work

PaintNinja committed:
Add dummy data

PaintNinja committed:
Ensure HTTPS for scripts, fixes local testing

PaintNinja committed:
Temporarily remove .gitignore

Want to make sure I haven't missed any dummy data

PaintNinja committed:
Add missed dummy data

PaintNinja committed:
Avoid people accidentally editing the wrong files

By having help text files and removing everything else from the out and build folders, we avoid people accidentally committing to the wrong things

PaintNinja committed:
Copy images folder over to inside the static folder

PaintNinja committed:
Add gitignore back in

PaintNinja committed:
Upgrade Gradle from 4.4.1 to 5.6.4

LexManos commented:
This needs rebase, we're discussing this on the discord, hardcoding test data is a bad idea. Thats 100+ files in the repo for no particularly good reason. So i added test file generation to the gradle

PaintNinja committed:
Merge branch 'MinecraftForge:master' into document-and-fix

PaintNinja committed:
Remove included dummy data

PaintNinja committed:
Flesh out Gradle script

- New tasks: setupPageGen and runTestPageGen

- Fix all digests being SHA-1
- Speed up hashing
- Explicitly type some things
- Fix typo
- Include recommended/latest builds in tests
- Include builds for both 1.18.1 and 1.18.2 in test data for page sidebar
- Move Help.txt over to test folder
- Update .gitignore to ignore test folder except Help.txt

PaintNinja committed:
Update instructions, remove more dummy data

PaintNinja committed:
Update Instructions.md

PaintNinja committed:
Add adfocus to dummy data

PaintNinja committed:
Merge branch 'master' of https://github.com/MinecraftForge/Web into MinecraftForge-master

PaintNinja committed:
*nix command tested

LexManos commented:
Not a fan of the powershell stuff would prefer to use a gradle python plugin.  but.. Thats a different task.
